specifyweb/frontend/js_src/lib/components/PickLists/TreeLevelPickList.tsx

```diff
@@ -96,7 +96,11 @@ export function TreeLevelComboBox(props: DefaultComboBoxProps): JSX.Element {
                 )
               : undefined
           )
-          .then((items) => (destructorCalled ? undefined : setItems(items))),
+          .then((items) => {
+            if (destructorCalled) return undefined;
+            resource.set('definitionItem', items?.slice(-1)[0]?.value || '');
+            return void setItems(items);
```

Comment:
sorry about `return void`. I will disable the eslint rule that forces that (as soon as xml-editor is out)